File: Employee-Management-Frontend/src/service/EmployeeService.js
import axios from "axios";

const BASE_URL = "http://localhost:1200/employee";

class EmployeeService {
  addEmployee(emp) {
    return axios.post(BASE_URL + "/addEmployee", emp);
  }

  getAllEmployee() {
    return axios.get(BASE_URL + "/allEmployee");
  }

  getEmployeeById(id) {
    return axios.get(BASE_URL + "/getEmployee/" + id);
  }

  deleteById(id) {
    return axios.delete(BASE_URL + "/deletEmployee/" + id);
  }

  updateEmployee(id, emp) {
    return axios.put(BASE_URL + "/updateEmployee/" + id, emp);
  }
}
export default new EmployeeService();

File: Employee-Management-Backend/src/main/java/com/jt/employee_management_backend/service/EmployeeService.java
package com.jt.employee_management_backend.service;

import java.util.List;

import com.jt.employee_management_backend.model.Employee;

public interface EmployeeService {

    public Employee addEmployee(Employee employee);

    public List<Employee> getAllEmployee();

    public Employee getEmployeeById(int id);

    public void deleteEmployee(int id);

    public Employee updateEmployee(int id, Employee employee);
}
